src/public/services/__tests__/AdminSubmissionsService.test.ts

```diff
@@ -0,0 +1,165 @@
+import axios from 'axios'
+import { mocked } from 'ts-jest/utils'
+
+import { SubmissionMetadataList } from 'src/types'
+
+import { ADMIN_FORM_ENDPOINT } from '../AdminFormService'
+import {
+  countFormSubmissions,
+  getEncryptedResponse,
+  getFormMetadata,
+  getFormsMetadata,
+} from '../AdminSubmissionsService'
+
+jest.mock('axios')
+const MockAxios = mocked(axios, true)
+
+describe('AdminSubmissionsService', () => {
+  describe('countFormSubmissions', () => {
+    const MOCK_FORM_ID = 'mock–form-id'
+    const MOCK_START_DATE = new Date(2020, 11, 17)
+    const MOCK_END_DATE = new Date(2021, 1, 10)
+
+    it('should call api successfully when all parameters are provided', async () => {
+      // Arrange
+      MockAxios.get.mockResolvedValueOnce({ data: 123 })
+
+      // Act
+      const actual = countFormSubmissions({
+        formId: MOCK_FORM_ID,
+        dates: { startDate: MOCK_START_DATE, endDate: MOCK_END_DATE },
+      })
+
+      // Assert
+      await expect(actual).resolves.toEqual(123)
+      expect(MockAxios.get).toHaveBeenCalledWith(
+        `${ADMIN_FORM_ENDPOINT}/${MOCK_FORM_ID}/submissions/count`,
+        {
+          params: {
+            startDate: MOCK_START_DATE,
+            endDate: MOCK_END_DATE,
+          },
+        },
+      )
+    })
+
+    it('should call api successfully when only formId is provided', async () => {
+      // Arrange
+      MockAxios.get.mockResolvedValueOnce({ data: 123 })
+
+      // Act
+      const actual = countFormSubmissions({
+        formId: MOCK_FORM_ID,
+      })
+
+      // Assert
+      await expect(actual).resolves.toEqual(123)
```

Comment:
like what anti has mentioned, since we don't use mockaxios anymore, we can await in the Act block instead of in the assert block

Reply:
update all the tests with the await in the assert block pattern too